vespa_feed_perf/src/main/java/com/yahoo/vespa/feed/perf/SimpleFeeder.java

```diff
@@ -59,31 +64,49 @@ public static void main(String[] args) throws Throwable {
         this.out = params.getStdOut();
         this.err = params.getStdErr();
         this.route = params.getRoute();
+        this.numThreads = params.getNumDispatchThreads();
         this.mbus = newMessageBus(docTypeMgr, params.getConfigId());
         this.session = newSession(mbus, this, params.isSerialTransferEnabled());
         this.docTypeMgr.configure(params.getConfigId());
     }
 
+    private void sendOperation(VespaXMLFeedReader.Operation op) {
+        Message msg = newMessage(op);
+        if (msg == null) {
+            err.println("ignoring operation; " + op.getType());
+            return;
+        }
+        msg.setContext(System.currentTimeMillis());
+        msg.setRoute(route);
+        try {
+            Error err = session.sendBlocking(msg).getError();
+            if (err != null) {
+                failure.set(new IOException(err.toString()));
+            }
+        } catch (InterruptedException e) {}
+    }
+
     SimpleFeeder run() throws Throwable {
+        ExecutorService executor = (numThreads > 1)
+                ? new ThreadPoolExecutor(numThreads, numThreads, 0L, TimeUnit.SECONDS,
```

Comment:
This looks equivalent to `Executors.newFixedThreadPool()`( https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/Executors.html#newFixedThreadPool(int,java.util.concurrent.ThreadFactory) )

Reply:
You can not control the Queue and the Policy of failed execution there.